src/main/java/com/LuuQu/medicalclinic/service/FacilityService.java

```diff
@@ -0,0 +1,52 @@
+package com.LuuQu.medicalclinic.service;
+
+import com.LuuQu.medicalclinic.mapper.FacilityMapper;
+import com.LuuQu.medicalclinic.model.dto.FacilityDto;
+import com.LuuQu.medicalclinic.model.entity.Facility;
+import com.LuuQu.medicalclinic.repository.FacilityRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class FacilityService {
+    private final FacilityRepository facilityRepository;
+    private final FacilityMapper facilityMapper;
+
+    public List<FacilityDto> getFacilities() {
+        return facilityRepository.findAll().stream()
+                .map(facilityMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    public FacilityDto getFacility(Long id) {
+        return facilityMapper.toDto(facilityRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("Non-existent facility")));
+    }
+
+    @Transactional
+    public FacilityDto addFacility(FacilityDto facilityDto) {
```

Comment:
w FacilityDto osoba, ktora dodaje nowa placowke nie wie jakie powinien byc jej id. 
W momencie jak wywołujesz save to save zauważ, że zwraca ci Encję, która została zapisana -> więc ta encja bedzie miala uzupełniony id i powinnismy znowu przemapować tą encję zwróconą przez save repository
return mapper.toDto(facilityRepository.save(facility));